kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/UserTaskServiceBase.java

```diff
@@ -86,6 +87,33 @@ public void claim(String containerId, Number taskId, String userId) {
     public void complete(String containerId, Number taskId, String userId, String payload, String marshallerType) {
 
         userId = getUser(userId);
+        
+        logger.debug("About to unmarshal task outcome parameters from payload: '{}'", payload);
+        Map<String, Object> parameters = marshallerHelper.unmarshal(containerId, payload, marshallerType, Map.class);
+
+        logger.debug("About to complete task with id '{}' as user '{}' with data {}", taskId, userId, parameters);
+        userTaskService.complete(taskId.longValue(), userId, parameters);
+
+    }
+
+    public void claimStartComplete(String containerId, Number taskId, String userId, String payload, String marshallerType) {
+
+        userId = getUser(userId);
+        Task task = userTaskService.getTask(taskId.longValue());
+        if (task.getTaskData().getStatus() == Status.Ready) {
+            
+            logger.debug("About to claim task automatically with id '{}' and user '{}'", taskId, userId);
+            userTaskService.claim(taskId.longValue(), userId);
+            logger.debug("About to start task automatically with id '{}' and user '{}'", taskId, userId);
+            userTaskService.start(taskId.longValue(), userId);
+            
+        } else if (task.getTaskData().getStatus() == Status.Reserved) {
+            
+            logger.debug("About to start task automatically with id '{}' and user '{}'", taskId, userId);
+            userTaskService.start(taskId.longValue(), userId);
+            
+        }
+        
```

Comment:
the issue with this code is that it creates several transactions:
- first to get task
- second to claim
- third to start
- forth to complete

of course it depends on the state of the task but still in worse case it will create and complete 4 transactions which is not efficient. I'd prefer to move this to jbpm services as this kind of functionality would make sense in embedded uses cases too.